onnxscript/rewriter/pattern.py

```diff
@@ -330,59 +330,53 @@ def pattern(x, shape1, shape2):
             t2 = op.Reshape(t1, shape2)
             return t2
     The above pattern matches a sequence of two Reshape ops.
-    The matched_values will contain the values representing the (output of)
-    the two Reshape ops, and the bindings will contain the values that
-    are bound to the variables `x`, `shape1`, and `shape2`.
+    The matched_nodes will contain the two Reshape ops, and the bindings will
+    contain the values that are bound to the variables `x`, `shape1`, and `shape2`.
     """
 
-    def __init__(
-        self, matched_values=None, bindings: dict[str, ir.Value | Any] | None = None
-    ) -> None:
-        assert matched_values is None or isinstance(matched_values, list)
-        self.success: bool = matched_values is not None
-        # For a successful match, matched_values is a list of values that matched the pattern.
+    def __init__(self, success: bool) -> None:
+        self.success: bool = success
+        # For a successful match, matched_nodes is a list of values that matched the pattern.
         # These include the internal nodes of the pattern that were matched, but not
         # the leaves (sub-trees) that match against the variables in the pattern.
         # These represent the values that will be replaced by the replacement pattern.
-        self.matched_values: Sequence[Any] | None = matched_values
+        self.matched_nodes: Sequence[ir.Node] = []
         # For a successful match, bindings is a dictionary of mapping pattern-variable-names
         # to values.
-        self.bindings: dict[str, Any] = bindings if bindings is not None else {}
+        self.bindings: dict[str, Any] = {}
 
     def __bool__(self):
         return self.success
 
     @classmethod
     def FAIL(cls):
-        return cls(None)
+        return cls(False)
 
     @property
-    def values(self) -> Sequence[Any] | None:
-        return self.matched_values
+    def nodes(self) -> Sequence[ir.Node]:
```

Comment:
```suggestion
    def nodes(self) -> MutableSequence[ir.Node]:
```

since we can append to it